File: leetcode/group2/637.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def averageOfLevels(self, root: TreeNode) -> List[float]:
        if not root:
            return 0
        queue = [root]
        ret = []

        while queue:
            n = len(queue)
            sum_ = 0
            for _ in range(n):
                node = queue.pop(0)
                sum_ += node.val
                if node.left:
                    queue.append(node.left)
                if node.right:
                    queue.append(node.right)
            ret.append(sum_/n)

        return ret

File: leetcode/花花酱题集/637.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number[]}
 */
var averageOfLevels = function (root) {
  if (!root) return 0;

  const queue = [root];
  const ans = [];

  while (queue.length > 0) {
    const n = queue.length;

    let sum = 0;
    for (let i = 0; i < n; i++) {
      const node = queue.shift();
      const { left, right, val } = node;
      sum += val;
      if (left) queue.push(left);
      if (right) queue.push(right);
    }
    ans.push(sum / n);
  }
  return ans;
};
